### PR TITLE
Unify existing search configuration into a single UI

### LINKED ISSUE #1174: Add Admin UI for configuring search fields
Assume we will enable the admin to:

- Turn on/off the fielded search option
- Configure the fields shown in the fielded search select

### LINKED ISSUE #1187: Update default home page text
The default homepage text block includes a link to "Curation > Search facets." However, as part of #1174  the Curation sidebar links will no longer include a link to "Search facets." They will include a link to "Search." 

To reflect that update, when #1174 is completed step 4 of the default home page text block should be updated. The text "Curation > Search facets" should be changed to "Curation > Search" and the link for that text should be updated to the new path for the Curation > Search page.

the_edward_a__feigenbaum_papers_-_stanford_university_libraries

### PR DESCRIPTION
**Pull request by cbeer**
This pull request merges sort, facet, and search results appearance configuration into a single search results interface. 

Fixes #1174 , fixes #1187

----

- [x] create search configurations controller, views, etc
- [x] add facet configuration
- [x] add results configuration
- [x] remove old configuration endpoints
  - [x] sort configurations
  - [x] facet configurations
  - [x] appearance configurations
- [x] fix default home page text (#1187)

~~add field search configuration :point_right: #1193~~
~~add 'check/uncheck all facets' option to facet tab :point_right:  #1194~~
~~remove 'searchable' boolean from exhibit, instead key off presence of search/facet fields to display features :point_right:  #1195~~

### CODE REVIEW

**Comment by ndushay** [comment]
process question:  why do we have a PR if we aren't ready to merge?

**Comment by cbeer** [comment]
I wanted somewhere to stick that checklist.. because this PR spans several tickets (and non-ticketed work), attaching it to the branch seemed useful to me.

File: spec/models/spotlight/blacklight_configuration_spec.rb [hunk 1]
```diff
@@ -450,4 +450,40 @@
       end
     end
   end
+
+  describe 'document_index_view_types=' do
```
Comment (cbeer) [inline spec/models/spotlight/blacklight_configuration_spec.rb]: Can we change `document_index_view_types=` to `#document_index_view_types`? It seems like we're testing both getter + setters.

File: spec/models/spotlight/blacklight_configuration_spec.rb [hunk 2]
```diff
@@ -450,4 +450,40 @@
...
+    end
+
+    it 'Hash should be simplified to Array' do
+      checkboxes_from_form = { 'list' => '1', 'gallery' => '1', 'map' => '0' }
+      subject.document_index_view_types = checkboxes_from_form
+      expect(subject.document_index_view_types).to match_array %w(list gallery)
+    end
+  end
+
+  describe 'document_index_view_types_selected_hash' do
```
Comment (cbeer) [inline spec/models/spotlight/blacklight_configuration_spec.rb]: Can we rename `document_index_view_types_selected_hash` to `#document_index_view_types_selected_hash` for consistency with rspec naming conventions?

File: spec/models/spotlight/blacklight_configuration_spec.rb [hunk 3]
```diff
@@ -450,4 +450,40 @@
       end
     end
   end
+
+  describe 'document_index_view_types=' do
+    it 'Array should pass through' do
```
Comment (cbeer) [inline spec/models/spotlight/blacklight_configuration_spec.rb]: Can we rephrase this to: `passes arrays through` (and below, `is simplified from a hash to an array`)?

**Comment by ndushay** [comment]
@cbeer help!  I copied the sort config view spec over, but I couldn't figure out how to get it to pass.  All other sort cleanup is fine -- any tests have been moved and other sort traces have been removed.

**Comment by ndushay** [comment]
I think this is ready to be merged!   My only question is if we neglected to test anything (perhaps something that wasn't formerly tested, but should have been - i think we got all the previous tests brought over)

**Comment by cbeer** [comment]
Squashed.

**Comment by cbeer** [comment]
Looks good to me. I reviewed the decrease in coverage, and I don't think it's because we missed tests, but just the number of well-covered features we were able to remove.
